File: cSharpe/Lookup.cs
using System;
using System.Collections.Generic;
using Twilio;
using Twilio.Rest.Lookups.V1;

namespace lookup
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("+ Lookup a phone number's data.");
            string accountSid = Environment.GetEnvironmentVariable("ACCOUNT_SID");
            string authToken = Environment.GetEnvironmentVariable("AUTH_TOKEN");
            Console.WriteLine("+ accountSid: " + accountSid);
            Console.WriteLine("+ authToken:  " + authToken);
            TwilioClient.Init(accountSid, authToken);
            var type = new List<string> {
                "carrier"
            };
            var phoneNumber = PhoneNumberResource.Fetch(
                type: type, // Include "type" to get carrier data.
                pathPhoneNumber: new Twilio.Types.PhoneNumber("+15108675310")
            );
            Console.WriteLine("++ phoneNumber.NationalFormat: " + phoneNumber.NationalFormat);
            Console.WriteLine("++ phoneNumber.CountryCode: " + phoneNumber.CountryCode);
            Console.WriteLine("++ Carrier: " + phoneNumber.Carrier);
        }
    }
}

File: php/lookup.php
<?php
require __DIR__ . '/../../twilio-php-master/Twilio/autoload.php';
use Twilio\Rest\Client;
$twilio = new Client(getenv('ACCOUNT_SID'), getenv('AUTH_TOKEN'));
// $phoneNumber = getenv('PHONE_NUMBER_4');
$phoneNumber = '+16505552222';
echo '++ Lookup phone number: ' . $phoneNumber . ":\xA";
$carriers = $twilio->lookups->v1->phoneNumbers(
        $phoneNumber)->fetch(
                array("type" => "carrier")
                );
print("+ Carrier: " . $carriers->carrier . "\xA");
var_dump($carriers->carrier);
//
$carrierName = $twilio->lookups->v1->phoneNumbers(
        $phoneNumber)->fetch(
                array("type" => "caller-name")
                );
print("+ callerName: " . $carrierName->callerName . "\xA");
var_dump($carrierName->callerName);
?>
